core/src/test/java/com/linecorp/armeria/server/logging/structured/StructuredLoggingServiceTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.apache.thrift.protocol.TMessage;
+import org.apache.thrift.protocol.TMessageType;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.logging.ResponseLog;
+import com.linecorp.armeria.common.thrift.ApacheThriftCall;
+import com.linecorp.armeria.common.thrift.ApacheThriftReply;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.thrift.THttpService;
+import com.linecorp.armeria.service.test.thrift.main.HelloService;
+import com.linecorp.armeria.service.test.thrift.main.HelloService.hello_args;
+import com.linecorp.armeria.service.test.thrift.main.HelloService.hello_result;
+import com.linecorp.armeria.test.AbstractServerTest;
+
+public class StructuredLoggingServiceTest extends AbstractServerTest {
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Captor
+    private ArgumentCaptor<ApacheThriftRpcStructuredLog> recordCaptor;
+
+    @Mock
+    private StructuredLogWriter<ApacheThriftRpcStructuredLog> structuredLogWriter;
+
+    private static HelloService.Iface newClient() throws Exception {
+        String uri = "tbinary+" + uri("/hello");
+        return Clients.newClient(uri, HelloService.Iface.class);
+    }
+
+    @Override
+    protected void configureServer(ServerBuilder sb) {
+        sb.serviceAt("/hello", THttpService.of((HelloService.Iface) name -> "Hello " + name)
+                                           .decorate(s -> new StructuredLoggingService<>(
```

Comment:
How about using `StructuredLoggingService.newDecorator()` and making the constructor private?